Applied permissions to self._error_file

kevinbsilva opened the pull request:
As explained at the issue [#15946](https://github.com/apache/airflow/issues/15946) the Web UI does not show the log for tasks that failed when the owner of the DAG is not airflow or the DAG is being executed by another user (run_as_user parameter). When airflow is dumping the error to the temporary file, it opens the file for writing and raises a Permission Denied error, since it's created with restricted permissions. Any default user cannot write into the file, only read.

The NamedTemporaryFile created at the base_task_runner creates the file with the umask 600 and that module does not allow to pass custom permissions when creating the object. 

To avoid this problem, i've added a os.chmod after creating the NamedTemporaryFile at the base_task_runner module. After that, no Permission Denied is thrown when a task fails and the log is accessible through the Web UI.

No major changes were made, only the chmod being applied solved the problem.

Closes https://github.com/apache/airflow/issues/15946

kevinbsilva committed:
Applied permissions to self._error_file

Used os.chmod with 0o0777 umask to avoid Permission Denied errors when accessing the temporary error file.

potiuk reviewed (dismissed):
Proposed alternative solution with correcting airflow deployment, not airflow code.

kevinbsilva committed:
Changed os.chmod to os.chown

Instead of using os.chmod and applying permissions to the error file, we are setting the file owner according to the variable self.run_as_user. This would avoid setting hardcoded permissions for the log file while maintaining the correct permissions to the user responsible for the DAG to manipulate the log file.

kevinbsilva reviewed (commented):
Changed os.chmod to os.chown

kevinbsilva committed:
Added exception to avoid erros at the os.chown, if receive something other than a string

kevinbsilva committed:
Ordered imports in alphabetical order

kevinbsilva committed:
Adjusted code format to black

kevinbsilva committed:
Fixed ordering group of imports

kevinbsilva committed:
Removed blank line from import. Applied isort

uranusjr commented on airflow/task/task_runner/base_task_runner.py:
```diff
@@ -85,6 +86,12 @@ def __init__(self, local_task_job):
+            try:
+                os.chown(self._error_file.name, getpwnam(self.run_as_user).pw_uid, -1)
+            except Exception:
+                pass
```
This should either bubble the exception or log that the chown failed. Personally I’d prefer *not* catching the exception for now and fix UX issues when someone complains.

ashb replied:
```suggestion
            os.chown(self._error_file.name, getpwnam(self.run_as_user).pw_uid, -1)
```

ashb committed:
Update airflow/task/task_runner/base_task_runner.py
